### ISSUE TITLE
rc_verbs failure under valgrind - Use of uninitialised value of size 8

### ISSUE THREAD

**Issue by evgeny-leksikov** [openucx/ucx#4991]
```
2020-04-07T09:37:38.6101857Z [ RUN      ] rc_verbs/test_uct_pending.pending_ucs_ok_dc_arbiter_bug/0 <rc_verbs/mlx4_0:1>
2020-04-07T09:37:38.7535793Z ==8572== Use of uninitialised value of size 8
2020-04-07T09:37:38.7538728Z ==8572==    at 0x5AD35D6: ??? (in /scrap/azure/agent-06/AZP_WORKSPACE/1/s/build-test/src/uct/ib/.libs/libuct_ib.so.0.0.0)
2020-04-07T09:37:38.7539895Z ==8572==    by 0x490613F: ???
2020-04-07T09:37:38.7540968Z ==8572==    by 0x5AEEA72: uct_rc_verbs_iface_post_recv_common (rc_verbs_impl.h:54)
2020-04-07T09:37:38.7542143Z ==8572==    by 0x5AEEA72: uct_rc_verbs_iface_common_prepost_recvs (rc_verbs_iface.c:313)
2020-04-07T09:37:38.7543323Z ==8572==    by 0x5AEEAFC: uct_rc_verbs_iface_common_progress_enable (rc_verbs_iface.c:330)
2020-04-07T09:37:38.7544393Z ==8572==    by 0x67F4FD: uct_iface_progress_enable (uct.h:3063)
2020-04-07T09:37:38.7545624Z ==8572==    by 0x67F4FD: uct_test::entity::entity(resource const&, uct_iface_config*, uct_iface_params*, uct_md_config*) (uct_test.cc:771)
2020-04-07T09:37:38.7547728Z ==8572==    by 0x680032: uct_test::create_entity(unsigned long, ucs_status_t (*)(void*, uct_ep*, ucs_status_t), ucs_status_t (*)(void*, void*, unsigned long, unsigned int, unsigned long, unsigned long, void**), ucs_status_t (*)(void*, unsigned int, unsigned long, void const*, unsigned int, unsigned long, unsigned long, void const*), void*, void*) (uct_test.cc:623)
2020-04-07T09:37:38.7549666Z ==8572==    by 0x65ABE5: test_uct_pending::init() (test_pending.cc:32)
2020-04-07T09:37:38.7550789Z ==8572==    by 0x59A3EE: ucs::test_base::SetUpProxy() (test.cc:242)
2020-04-07T09:37:38.7552273Z ==8572==    by 0x57EE58: HandleSehExceptionsInMethodIfSupported<testing::Test, void> (gtest-all.cc:3562)
2020-04-07T09:37:38.7554186Z ==8572==    by 0x57EE58: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest-all.cc:3598)
2020-04-07T09:37:38.7555873Z ==8572==    by 0x572FFD: testing::Test::Run() (gtest-all.cc:3630)
2020-04-07T09:37:38.7557205Z ==8572==    by 0x573174: testing::TestInfo::Run() (gtest-all.cc:3812)
2020-04-07T09:37:38.7558570Z ==8572==    by 0x5732DE: testing::TestCase::Run() (gtest-all.cc:3930)
2020-04-07T09:37:38.7560080Z ==8572==    by 0x577A8C: testing::internal::UnitTestImpl::RunAllTests() [clone .part.343] (gtest-all.cc:5808)
2020-04-07T09:37:38.7561551Z ==8572==    by 0x577D6F: RunAllTests (gtest-all.cc:5725)
2020-04-07T09:37:38.7563075Z ==8572==    by 0x577D6F: HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (gtest-all.cc:3562)
2020-04-07T09:37:38.7564774Z ==8572==    by 0x577D6F: HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (gtest-all.cc:3598)
2020-04-07T09:37:38.7566274Z ==8572==    by 0x577D6F: testing::UnitTest::Run() (gtest-all.cc:5422)
2020-04-07T09:37:38.7567340Z ==8572==    by 0x516AD7: RUN_ALL_TESTS (gtest.h:20059)
2020-04-07T09:37:38.7568277Z ==8572==    by 0x516AD7: main (main.cc:102)
2020-04-07T09:37:38.7569202Z ==8572==  Uninitialised value was created by a stack allocation
2020-04-07T09:37:38.7570171Z ==8572==    at 0x8AE7700: ??? (in /usr/lib64/libmlx4.so.1.0.28.0)
2020-04-07T09:37:38.7571055Z ==8572== 
2020-04-07T09:37:38.7606754Z ==8572== WARNING: valgrind ignores shmget(shmflg) SHM_HUGETLB
2020-04-07T09:37:39.2171356Z [     SKIP ] (Can't fill UCT resources in the given time.)
2020-04-07T09:37:39.4066499Z /scrap/azure/agent-06/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test.cc:275: Failure
2020-04-07T09:37:39.4068081Z Failed
2020-04-07T09:37:39.4068868Z Got 1 valgrind errors during the test
2020-04-07T09:37:39.4300539Z [  FAILED  ] rc_verbs/test_uct_pending.pending_ucs_ok_dc_arbiter_bug/0, where GetParam() = rc_verbs/mlx4_0:1 (806 ms)
```

**Comment by alinask** [comment]
On swx-rdmz-ucx-legacy-01:

```
[ RUN      ] rc_verbs/uct_p2p_err_test.invalid_put_short_length/0 <rc_verbs/mlx4_0:1>
==26767== Use of uninitialised value of size 8
==26767==    at 0x5AD35D6: ??? (in /scrap/azure/agent-01/AZP_WORKSPACE/1/s/build-test/src/uct/ib/.libs/libuct_ib.so.0.0.0)
==26767==    by 0x3EA77CF44765195E: ???
==26767==    by 0x5AEEAA2: uct_rc_verbs_iface_post_recv_common (rc_verbs_impl.h:54)
==26767==    by 0x5AEEAA2: uct_rc_verbs_iface_common_prepost_recvs (rc_verbs_iface.c:313)
==26767==    by 0x5AEEB2C: uct_rc_verbs_iface_common_progress_enable (rc_verbs_iface.c:330)
==26767==    by 0x67EA2D: uct_iface_progress_enable (uct.h:3063)
==26767==    by 0x67EA2D: uct_test::entity::entity(resource const&, uct_iface_config*, uct_iface_params*, uct_md_config*) (uct_test.cc:771)
==26767==    by 0x67F562: uct_test::create_entity(unsigned long, ucs_status_t (*)(void*, uct_ep*, ucs_status_t), ucs_status_t (*)(void*, void*, unsigned long, unsigned int, unsigned long, unsigned long, void**), ucs_status_t (*)(void*, unsigned int, unsigned long, void const*, unsigned int, unsigned long, unsigned long, void const*), void*, void*) (uct_test.cc:623)
==26767==    by 0x67449E: uct_p2p_test::init() (uct_p2p_test.cc:68)
==26767==    by 0x59991E: ucs::test_base::SetUpProxy() (test.cc:242)
==26767==    by 0x57E388: HandleSehExceptionsInMethodIfSupported<testing::Test, void> (gtest-all.cc:3562)
==26767==    by 0x57E388: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest-all.cc:3598)
==26767==    by 0x57252D: testing::Test::Run() (gtest-all.cc:3630)
==26767==    by 0x5726A4: testing::TestInfo::Run() (gtest-all.cc:3812)
==26767==    by 0x57280E: testing::TestCase::Run() (gtest-all.cc:3930)
==26767==    by 0x576FBC: testing::internal::UnitTestImpl::RunAllTests() [clone .part.343] (gtest-all.cc:5808)
==26767==    by 0x57729F: RunAllTests (gtest-all.cc:5725)
==26767==    by 0x57729F: HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (gtest-all.cc:3562)
==26767==    by 0x57729F: HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (gtest-all.cc:3598)
==26767==    by 0x57729F: testing::UnitTest::Run() (gtest-all.cc:5422)
==26767==    by 0x516567: RUN_ALL_TESTS (gtest.h:20059)
==26767==    by 0x516567: main (main.cc:102)
==26767==  Uninitialised value was created by a stack allocation
==26767==    at 0x8CEC700: ??? (in /usr/lib64/libmlx4.so.1.0.28.0)
==26767== 
[     INFO ] < Invalid put_short length: 89 (expected: <= 88) >
/scrap/azure/agent-01/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test.cc:275: Failure
Failed
Got 1 valgrind errors during the test
[  FAILED  ] rc_verbs/uct_p2p_err_test.invalid_put_short_length/0, where GetParam() = rc_verbs/mlx4_0:1 (527 ms)

```

On swx-rdmz-ucx-legacy-02:
```
[ RUN ] rc_verbs/test_zcopy_comp.issue1440/0 <rc_verbs/mlx4_0:1> 
==13848== Use of uninitialised value of size 8 
==13848== at 0x5AD35D6: ??? (in /scrap/azure/agent-02/AZP_WORKSPACE/1/s/build-test/src/uct/ib/.libs/libuct_ib.so.0.0.0) 
==13848== by 0x3FEDCD5D3FFFFFFF: ??? 
==13848== by 0x5AEEAA2: uct_rc_verbs_iface_post_recv_common (rc_verbs_impl.h:54) 
==13848== by 0x5AEEAA2: uct_rc_verbs_iface_common_prepost_recvs (rc_verbs_iface.c:313) 
==13848== by 0x5AEEB2C: uct_rc_verbs_iface_common_progress_enable (rc_verbs_iface.c:330) 
==13848== by 0x67EA2D: uct_iface_progress_enable (uct.h:3063) 
==13848== by 0x67EA2D: uct_test::entity::entity(resource const&, uct_iface_config*, uct_iface_params*, uct_md_config*) (uct_test.cc:771) 
==13848== by 0x67F562: uct_test::create_entity(unsigned long, ucs_status_t (*)(void*, uct_ep*, ucs_status_t), ucs_status_t (*)(void*, void*, unsigned long, unsigned int, unsigned long, unsigned long, void**), ucs_status_t (*)(void*, unsigned int, unsigned long, void const*, unsigned int, unsigned long, unsigned long, void const*), void*, void*) (uct_test.cc:623) 
==13848== by 0x671BAC: test_zcopy_comp::init() (test_zcopy_comp.cc:14) 
==13848== by 0x59991E: ucs::test_base::SetUpProxy() (test.cc:242) 
==13848== by 0x57E388: HandleSehExceptionsInMethodIfSupported<testing::Test, void> (gtest-all.cc:3562) 
==13848== by 0x57E388: void testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test*, void (testing::Test::*)(), char const*) (gtest-all.cc:3598) 
==13848== by 0x57252D: testing::Test::Run() (gtest-all.cc:3630) 
==13848== by 0x5726A4: testing::TestInfo::Run() (gtest-all.cc:3812) 
==13848== by 0x57280E: testing::TestCase::Run() (gtest-all.cc:3930) 
==13848== by 0x576FBC: testing::internal::UnitTestImpl::RunAllTests() [clone .part.343] (gtest-all.cc:5808) 
==13848== by 0x57729F: RunAllTests (gtest-all.cc:5725) 
==13848== by 0x57729F: HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (gtest-all.cc:3562) 
==13848== by 0x57729F: HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool> (gtest-all.cc:3598) 
==13848== by 0x57729F: testing::UnitTest::Run() (gtest-all.cc:5422) 
==13848== by 0x516567: RUN_ALL_TESTS (gtest.h:20059) 
==13848== by 0x516567: main (main.cc:102) 
==13848== Uninitialised value was created by a stack allocation 
==13848== at 0x8AE7700: ??? (in /usr/lib64/libmlx4.so.1.0.28.0) 
==13848== 
==13848== WARNING: valgrind ignores shmget(shmflg) SHM_HUGETLB 
/scrap/azure/agent-02/AZP_WORKSPACE/1/s/contrib/../test/gtest/common/test.cc:275: Failure 
Failed 
```

**Comment by dmitrygx** [comment]
@Artemy-Mellanox @yosefe looks like it has already been fixed in the master, right?

**Comment by yosefe** [comment]
fixed in master and v1.8.x
